nlpcore/src/latin/prediction.cppm

```diff
@@ -97,21 +112,47 @@ struct RecursiveFuzzySearchDistanceCell {
     bool is_equal_ignoring_case_ = false;
 };
 
+double computeConfidence(bool isWordPrefix, float cost, float frequency, std::size_t word_size, size_t token_size) {
+    // TODO: reevaluate the weighting and calculation
+    double w1 = 1.0;
+    double w2 = 0.5;
+    double similarity;
+    // if (isWordPrefix) {
+    //     similarity = 1.0 - (cost / std::max(static_cast<std::size_t>(1), word_size));
+    // } else {
+    //     similarity = 1.0 - (cost / std::max(token_size, word_size));
+    // }
+    similarity = -cost;
+    double confidence = (w1 * similarity + w2 * frequency) / (w1 + w2);
```

Comment:
Why assign negative cost to similarity?

Reply:
It's from the "just get it working" mindset. The similarity does not really need to be 0-1 for suggestion; it just needs to be correctly ordered such that better matches are scored higher. I'll work out the math in the next update.